src/ucp/core/ucp_context.c

```diff
@@ -666,6 +669,31 @@ static ucs_status_t ucp_fill_resources(ucp_context_h context,
          * client-server connection establishment via sockaddr, don't use it */
         if ((num_tl_resources > 0) ||
             (context->tl_mds[md_index].attr.cap.flags & UCT_MD_FLAG_SOCKADDR)) {
+            /*List of memory type MDs*/
+            if (context->tl_mds[md_index].attr.cap.mem_type != UCT_MD_MEM_TYPE_HOST) {
+                if (context->mem_type_tl_mds == NULL) {
```

Comment:
my point was that this allocation can be done *before* going over mds.. in line 667 for example

Reply:
moved it to before loop over the mds. It is just that allocation may be unused if there is no memory types.